api/src/main/java/com/epam/pipeline/manager/docker/DockerParsingUtils.java

```diff
@@ -93,6 +106,62 @@ public static Map<String, String> getLabels(final RawImageDescription rawImage)
                 .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
     }
 
+    public static List<String> processCommands(final String from, final List<String> commands,
+                                               final List<OSSpecificLaunchCommandTemplate> podLaunchTemplatesLinux,
+                                               final String podLaunchTemplatesWin) {
+        final List<String> result = new ArrayList<>();
+        result.add(String.format(FROM_TEMPLATE, from));
+        if (CollectionUtils.isEmpty(commands)) {
+            return result;
+        }
+        final List<String> podLaunchPatterns = getPodLaunchPatterns(podLaunchTemplatesLinux, podLaunchTemplatesWin);
+        final List<Integer> cmdIndexes = new ArrayList<>();
+        final List<Integer> entrypointIndexes = new ArrayList<>();
+        final List<String> args = new ArrayList<>();
+        for (int i = 0; i < commands.size(); i++) {
+            String command = commands.get(i);
+            if (command.startsWith(ARG)) {
+                args.add(command.replace(ARG, ""));
+            } else if (command.startsWith(CMD)) {
+                cmdIndexes.add(i);
+            } else if (command.startsWith(ENTRYPOINT)) {
+                entrypointIndexes.add(i);
+            }
+        }
+        final int lastCmdIndex = CollectionUtils.isEmpty(cmdIndexes) ? -1 : cmdIndexes.get(cmdIndexes.size() - 1);
+        final int lastEntrypointIndex = CollectionUtils.isEmpty(entrypointIndexes) ? -1 :
+                entrypointIndexes.get(entrypointIndexes.size() - 1);
+
+        final int startIndex = commands.get(0).startsWith(ADD_FILE) ? 1 : 0;
```

Comment:
I would move this line closer to [FROM](https://github.com/epam/cloud-pipeline/pull/3572/files#diff-92af65798e97542bc9df5de528071ff150788a2b937522d289cdad576cf881a7R113) and leave a comment why we are doing that